Homework2/Lazy/Lazy/SingleThreadLazy.cs

```diff
@@ -0,0 +1,52 @@
+namespace Lazy;
+
+/// <summary>
+/// Class that represents single thread version of Lazy interface.
+/// </summary>
+/// <typeparam name="T">Generic type of result of calculating.</typeparam>
+public class SingleThreadLazy<T> : ILazy<T>
+{
+    private Func<T>? supplier;
```

Comment:
```suggestion
    private Func<T?>? supplier;
```
функция должна уметь возвращать null, верно ведь?